### PR TITLE
Implemented Initializing status.

### PR DESCRIPTION
**Pull request by sfmskywalker**
Issue: #5650

This PR adds an additional `Initializing` status to the `TenantState` enum and provides an appropriate response to the browser in case a request is made while the tenant is in the `Initializing` state.

### CODE REVIEW

**Comment by Jetski5822** [comment]
Ill need to remember to bring this across to Brochard!

**Comment by sfmskywalker** [comment]
Yeah, and not just this I suppose. I imagine there'll be some other things that change over time.

File: src/Orchard/Environment/ShellBuilders/ShellContextFactory.cs [hunk 1]
```diff
@@ -103,7 +103,8 @@ public class ShellContextFactory : IShellContextFactory {
                 Features = new[] {
                     new ShellFeature { Name = "Orchard.Setup" },
                     new ShellFeature { Name = "Shapes" },
-                    new ShellFeature { Name = "Orchard.jQuery" },
+                    new ShellFeature { Name = "Orchard.Recipes" },
+                    new ShellFeature { Name = "Orchard.jQuery" }
```
Comment (sebastienros) [inline src/Orchard/Environment/ShellBuilders/ShellContextFactory.cs]: How is it related to the PR ?

Comment (sfmskywalker) [inline src/Orchard/Environment/ShellBuilders/ShellContextFactory.cs]: It's not actually, good catch. It's a leftover of some of the refactoring I did but forgot to undo this. I'll get rid of it.

**Pull request merged**